inprocess/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -414,6 +418,31 @@ private void streamClosed() {
       }
     }
 
+    private int getKnownLength(InputStream inputStream) throws IOException {
+      if (inputStream instanceof KnownLength || inputStream instanceof ByteArrayInputStream) {
+        return inputStream.available();
+      }
+      return -1;
+    }
+
+    private long getMessageLength(InputStream inputStream) {
+      try {
+        return getKnownLength(inputStream);
+      } catch (IOException e) {
+        throw Status.INTERNAL
+                .withDescription("Failed to calculate size")
+                .withCause(e)
+                .asRuntimeException();
+      } catch (StatusRuntimeException e) {
```

Comment:
We don't need to special-case StatusRuntimeException. This code won't throw that, whereas in MessageFramer (which it seems you were copying), the code would use such an exception itself. If an InputStream throws a StatusRuntimeException, I think we still want to use our status code.

That also means we can use `IOException|RuntimeException` syntax now.